docs/sbom-protocol.md

```diff
@@ -0,0 +1,72 @@
+# SBOM Scanning Protocol
+
+BuildKit supports automatic creation of [SBOMs](https://en.wikipedia.org/wiki/Software_supply_chain)
+for builds, attaching them as [image attestations](./attestation-storage.md).
+    
+To scan the filesystem contents, a user can specify an SBOM generator image.
+When run, this image is passed the rootfs of the build stage as a read-only
+mount, writes its SBOM scan data to a specified directory.
+
+The SBOM generator image is expected to follow the rules of the BuildKit SBOM
+generator protocol, defined in this document.
+
+> **Note**
+>
+> Currently, only SBOMs in the [SPDX](https://spdx.dev) JSON format are
```

Comment:
Follow-up: could we use some labels to track what is supported so we have a upgrade path?

Reply:
I assume you mean in the image labels? Yup, I'll do a follow-up :smile: